draft-irtf-cfrg-hash-to-curve.md

```diff
@@ -1581,75 +1581,141 @@ Steps:
 26. return (x, y)
 ~~~
 
-## Mappings for twisted Edwards curves
-
-### Elligator 2 Method {#ell2edwards}
-
-Twisted Edwards curves and Montgomery curves are closely related: every
-twisted Edwards curve is birationally equivalent to a Montgomery curve
-({{BBJLP08}}, Theorem 3.2). To hash to a twisted Edwards curve, hash
-to the equivalent Montgomery curve and evaluate the rational map to
-obtain a point on the twisted Edwards curve.
-
-For a twisted Edwards curve given by a * x^2 + y^2 = 1 + d * x^2 * y^2,
-first compute A and B, the parameters of the equivalent Montgomery curve,
-as follows:
-
-- A = (a + 2) / 2
+## Mappings for Edwards curves
+
+Edwards curves are closely related to Montgomery
```

Comment:
Why it was removed 'twisted'?
Twisted Edwards are more generic than Edwards curves.

Reply:
You're absolutely right, I was sloppy. Thanks for catching!

Reply:
To address this, I added a parenthetical clarifying that we say "Edwards" to mean both twisted (a != 1) and untwisted (a = 1).

This isn't a perfect solution, because Bernstein et al. (in the paper that introduced Twisted Edwards Curves, BBJLP08) use "Edwards curve" to mean "a = 1" (i.e., untwisted), and "twisted Edwards curve" to refer to the more general case that a may not be equal to 1.

The alternative, which is strictly correct but probably much more annoying to read, is to use the term "twisted Edwards curve" everywhere in the document. To me, this likely *reduces* the clarity of the document.

I'd rather annoy the small handful of people who both understand and care about the above distinction than confuse the much larger number of people who don't, but I understand the argument either way.

Reply:
I prefer to name curves by their canonical name. As you can see in RFC7748 and RFC8032, the full name `twisted Edwards curves` is used without confusion.

When referring to a twisted Edwards curve, they are also included the Edwards curves (a=1). So, no distinction must be done, since the birational map applies generally.

Reply:
OK, I just pushed a commit that writes "twisted Edwards" everywhere.

I suggest that each of us compares the clarity with and without this commit. If it's clearer to slightly abuse terminology, is that extra clarity worth the abuse? Alternatively, maybe my thoughts on clarity above are a bit alarmist, and we should just use the correct terminology.

I'm happy to go with whichever one people like.

Reply:
OK, maybe I had too much caffeine earlier. Happy to stick with "twisted" everywhere :)